OCPBUGS-43331: update release note for 4.15.18 (#83474)
openshift/openshift-docs: release_notes/ocp-4-15-release-notes.adoc

The following enhancements are included in this z-stream release:

[id="ocp-4-15-18-unservable-future-versions-status-condition"]
===== Blocking upgrades for SHA1 serving certificates on routes and Ingress controllers

> * {product-title} {product-version} does not support SHA1 certificates on routes. Consequently, upgrades from 4.15 to 4.16 cause routes and Ingress controllers with SHA1 certificates to be rejected.

Reviewer: The intention is to indicate that the next version (4.16) of OpenShift doesn't support SHA1 certs. This version, 4.15 does. I believe `{product-version}` substitutes to 4.15. Is there a way to do `{product-version+1}`?

```suggestion
* {product-title} 4.16 does not support SHA1 certificates on routes. Consequently, upgrades from 4.15 to 4.16 cause routes and Ingress controllers with SHA1 certificates to be rejected.
```

> * {product-title} {product-version} does not support SHA1 certificates on routes. Consequently, upgrades from 4.15 to 4.16 cause routes and Ingress controllers with SHA1 certificates to be rejected.

Reviewer 2: This release note relates to IngressControllers objects, not controllers for Ingress objects, so it should be written "Ingress Controllers" per the glossary.  (Naming is hard.)  This comment applies to all 4 occurrences (once in the heading and thrice in the text under the heading).  
https://github.com/openshift/openshift-docs/blob/230e6dd459166b0b3b7af998bd059025e166f238/contributing_to_docs/term_glossary.adoc?plain=1#L377-L382

+
Only serving certificates are affected. For routes, this is the certificate specified in `spec.tls.certificate`. For Ingress controllers, this is the serving certificate in the secret specified in `spec.defaultCertificate`. CA certificates using SHA1 are not impacted. However, due to link:https://issues.redhat.com/browse/OCPBUGS-42480[*OCPBUGS-42480*], some Ingress controllers were incorrectly blocking upgrades if the serving certificate was not using SHA1, but the CA certificate was. To resolve this, upgrade to a 4.15 version with this bug fix.
+

> Additionally, this update introduces a new `UnservableInFutureVersions` status condition to routes that contain a SHA1 certificate. It also adds an `admin-gate` to block upgrades if this new status is present on any route. As a result, if cluster administrators have routes that use SHA1 certificates in {product-title} 4.15, they must either upgrade these certificates to a support algorithm or provide an `admin-ack` for the created `admin-gate`. This `admin-ack` allows administrators to proceed with the upgrade without resolving the SHA1 certificate issues, even though the routes will be rejected. (link:https://issues.redhat.com/browse/OCPBUGS-28928[*OCPBUGS-28928*]).

Reviewer 2: Typo: "suggest" → "suggested".
```suggestion
Additionally, this update introduces a new `UnservableInFutureVersions` status condition to routes that contain a SHA1 certificate. It also adds an `admin-gate` to block upgrades if this new status is present on any route. As a result, if cluster administrators have routes that use SHA1 certificates in {product-title} 4.15, they must either upgrade these certificates to a supported algorithm or provide an `admin-ack` for the created `admin-gate`. This `admin-ack` allows administrators to proceed with the upgrade without resolving the SHA1 certificate issues, even though the routes will be rejected. (link:https://issues.redhat.com/browse/OCPBUGS-28928[*OCPBUGS-28928*]).
```

Reviewer 2: @gcs278, where do we document the command to ack the gate?  Should be `oc -n openshift-config patch cm admin-acks --patch '{"data":{"ack-4.15-route-config-not-supported-in-4.16":"true"}}' --type=merge`, right?

Reviewer: I don't see any product documentation. We just have the enhancement https://github.com/openshift/enhancements/blob/master/enhancements/update/upgrades-blocking-on-ack.md

It actually looks like this admin gate implementation was a bugzilla, and I see one comment about product documentation from Ben: https://bugzilla.redhat.com/show_bug.cgi?id=1978376#c11, but I don't see where it was done.

Do you think it's worth asking the updates team for product documentation?

The following enhancements are included in this z-stream release:

[id="ocp-4-15-18-unservable-future-versions-status-condition"]
===== Blocking upgrades for SHA1 serving certificates on routes and Ingress Controllers

> * {product-title} {product-version} does not support SHA1 certificates on routes. Consequently, upgrades from 4.15 to 4.16 cause routes and Ingress Controllers with SHA1 certificates to be rejected.

Reviewer: Does `{product-version}` get substituted to 4.15? I was thinking that 4.16 doesn't support SHA1 certs, while 4.15 still supports, but sets Upgradeable=False. Does this language matter?

+

> Only serving certificates are affected. For routes, this is the certificate specified in `spec.tls.certificate`. For Ingress Controllers, this is the serving certificate in the secret specified in `spec.defaultCertificate`. CA certificates using SHA1 are not impacted. However, due to link:https://issues.redhat.com/browse/OCPBUGS-42480[*OCPBUGS-42480*], some Ingress Controllers were incorrectly blocking upgrades if the serving certificate was not using SHA1, but the CA certificate was. To resolve this, upgrade to a 4.15 version with this bug fix.

Reviewer: ```suggestion
Only serving certificates are affected. For routes, this is the certificate specified in `spec.tls.certificate`. For Ingress Controllers, this is the serving certificate in the secret specified in `spec.defaultCertificate`. CA certificates using SHA1 are not impacted. However, due to link:https://issues.redhat.com/browse/OCPBUGS-42480[*OCPBUGS-42480*], some Ingress Controllers were incorrectly blocking upgrades if the serving certificate was not using SHA1, but the CA certificate was. To resolve this, upgrade to 4.15.37 to receive this bug fix.
```
